File: src/main/java/discretemath/bool/expression/atomic/AtomicBooleanExpression.java
package discretemath.bool.expression.atomic;

import discretemath.bool.expression.BooleanExpression;
import discretemath.bool.operator.FunctionallyCompleteOperatorsSet;

public sealed interface AtomicBooleanExpression extends BooleanExpression
		permits BooleanVariable, ConstantBooleanValue {

	@Override
	default BooleanExpression expressViaOperators(FunctionallyCompleteOperatorsSet operatorsSet) {
		return this;
	}

	@Override
	default BooleanExpression normalize() {
		return this;
	}
}

File: src/main/scala/discretemath/bool/expression/atomic/AtomicBooleanExpression.scala
package discretemath.bool.expression.atomic

import discretemath.bool.expression.BooleanExpression
import discretemath.bool.operator.FunctionallyCompleteOperatorsSet

trait AtomicBooleanExpression extends BooleanExpression {

  final override def expressViaOperators(operatorsSet: FunctionallyCompleteOperatorsSet): BooleanExpression = this

  final override def minimize: BooleanExpression = this
}
